agent/api/task.go

```diff
@@ -485,30 +490,50 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		}
 	}
 
-	// Set network mode in host config. Since we create internal containers for
-	// volumes and setting up networking for ENIs, we handle those special cases
-	// via a switch statement
-	switch container.Name {
-	// emptyHostVolumeName inidicates that this is an internal container
-	// and is used for setting up empty host volume mounts. Do not change
-	// its network mode
-	case emptyHostVolumeName:
-	// pauseContainerName indicates this is an internal container and is
-	// used for setting up the network namespace of the pause container.
-	// Such a container must be created with the "none" network mode
-	case pauseContainerName:
-		hostConfig.NetworkMode = networkModeNone
-	// default case for all other containers is to set their network mode
-	// to "container:<pause-container-id>" if the pause container is a
-	// part of the task's container list
-	default:
-		if pauseContainer, ok := dockerContainerMap[pauseContainerName]; ok {
-			hostConfig.NetworkMode = "container:" + pauseContainer.DockerID
-		}
+	// Determine if network mode should be overridden and override it if needed
+	ok, networkMode := task.shouldOverrideNetworkMode(container, dockerContainerMap)
+	if !ok {
+		return hostConfig, nil
 	}
+	hostConfig.NetworkMode = networkMode
 	return hostConfig, nil
 }
 
+// shouldOverrideNetworkMode returns true if the network mode of the container needs
+// to be overridden. It also returns the override string in this case. It returns
+// false otherwise
+func (task *Task) shouldOverrideNetworkMode(container *Container, dockerContainerMap map[string]*DockerContainer) (bool, string) {
```

Comment:
For this function, instead of check all the containers within the task, could we depend on a `field` in the `Task` struct eg: "ENIS" to determine if the network mode should be override. Since there will be some new filed specific to this feature adding to the `Task` struct: